publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/helper/RabbitMqProperties.java

```diff
@@ -272,15 +273,36 @@ public void createRabbitMqConnection() throws RemRemPublishException {
             }
             factory.setConnectionTimeout(tcpTimeOut);
             rabbitConnection = factory.newConnection();
+            rabbitConnection.addShutdownListener(new ShutdownListener() {
+                @Override
+                public void shutdownCompleted(ShutdownSignalException cause) {
+                    log.debug("Connection Shutdown completed " + cause);
+                    try {
+                        rabbitConnection.close();
+                    } catch (Throwable e) {
+                        // TODO Auto-generated catch block
+                        e.printStackTrace();
+                    }
+                }
+            });
+
+            rabbitConnection.addBlockedListener(new BlockedListener() {
+                public void handleBlocked(String reason) throws IOException {
+                    // Connection is now blocked
+                    log.debug("Connection is blocked " + reason);
```

Comment:
reason.getMessage()

Reply:
the variable reason is type of string, which returns the proper message.